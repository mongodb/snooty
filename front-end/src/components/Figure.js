import React, { Component } from 'react';

export default class Figure extends Component {

  render() {
    return (
      <img src={ this.props.nodeData.argument[0].value } alt={ this.props.nodeData.options.alt ? this.props.nodeData.options.alt : this.props.nodeData.argument[0].value } width="50%" />
    )
  }

}