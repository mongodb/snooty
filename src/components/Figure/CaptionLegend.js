import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { uiColors } from '@leafygreen-ui/palette';
import ComponentFactory from '../ComponentFactory';
import { theme } from '../../theme/docsTheme';

const Caption = styled('p')`
  color: ${uiColors.gray.dark1};
  /* TODO: Remove !important when mongodb-docs.css is removed */
  margin-top: ${theme.size.default} !important;
  text-align: center;

  /* TODO: Remove when mongodb-docs.css is removed */
  & > code {
    color: ${uiColors.gray.dark1};
  }
`;

const CaptionLegend = ({ nodeData: { children }, ...rest }) => (
  <>
    {children.length > 0 && (
      <Caption>
        <ComponentFactory {...rest} nodeData={children[0]} parentNode="caption" />
      </Caption>
    )}
    {children.length > 1 && (
      <>
        {children.slice(1).map((child, index) => (
          <ComponentFactory {...rest} key={index} nodeData={child} />
        ))}
      </>
    )}
  </>
);

CaptionLegend.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
};

export default CaptionLegend;
