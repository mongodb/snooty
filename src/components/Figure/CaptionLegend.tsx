import React from 'react';
import styled from '@emotion/styled';
import { palette } from '@leafygreen-ui/palette';
import ComponentFactory from '../ComponentFactory';
import { theme } from '../../theme/docsTheme';
import { FigureProps } from '.';

const Caption = styled('p')`
  color: ${palette.gray.dark1};
  /* TODO: Remove !important when mongodb-docs.css is removed */
  margin-top: ${theme.size.default} !important;
  text-align: center;

  /* TODO: Remove when mongodb-docs.css is removed */
  & > code {
    color: ${palette.gray.dark1};
  }
`;

const CaptionLegend = ({ nodeData: { children }, ...rest }: FigureProps) => (
  <>
    {children.length > 0 && (
      <Caption>
        <ComponentFactory {...rest} nodeData={children[0]} parentNode="caption" />
      </Caption>
    )}
    {children.length > 1 && (
      <>
        {children.slice(1).map((child, index) => (
          <ComponentFactory {...rest} key={index} nodeData={child} />
        ))}
      </>
    )}
  </>
);

export default CaptionLegend;
