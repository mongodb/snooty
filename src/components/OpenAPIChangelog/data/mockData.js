export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
