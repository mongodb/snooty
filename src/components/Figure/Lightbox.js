import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Modal from '@leafygreen-ui/modal';
import styled from '@emotion/styled';
import Image from '../Image';
import { theme } from '../../theme/docsTheme.js';
import CaptionLegend from './CaptionLegend';

const CAPTION_TEXT = 'click to enlarge';
const StyledModal = styled(Modal)`
  // Set z-index to appear above side nav and top navbar
  z-index: 10;
  margin-top: 64px;

  @media ${theme.screenSize.largeAndUp} {
    div[role='dialog'] {
      max-width: 80%;
    }
    img {
      max-height: 620px;
    }
  }

  @media ${theme.screenSize.upToLarge} {
    div[role='dialog'] {
      max-width: 100%;
    }
    img {
      max-height: 300px;
    }
  }
`;

const LightboxCaption = styled('div')`
  color: #444;
  font-size: 80%;
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  text-align: center;
`;

const LightboxWrapper = styled('div')`
  width: ${(props) => props.figwidth};
  cursor: pointer;
  margin-top: ${theme.size.medium};
  margin-bottom: ${theme.size.medium};
  display: block;
  max-height: 620px;
`;

const Lightbox = ({ nodeData, ...rest }) => {
  const [open, setOpen] = useState(false);
  const figureWidth = nodeData.options?.figwidth || 'auto';
  return (
    <React.Fragment>
      <LightboxWrapper figwidth={figureWidth}>
        <div onClick={() => setOpen((curr) => !curr)} role="button" tabIndex="-1">
          <Image nodeData={nodeData} />
          <LightboxCaption>{CAPTION_TEXT}</LightboxCaption>
        </div>
        <CaptionLegend {...rest} nodeData={nodeData} />
      </LightboxWrapper>
      <StyledModal size="medium" open={open} setOpen={setOpen}>
        <Image nodeData={nodeData} />
      </StyledModal>
    </React.Fragment>
  );
};

Lightbox.propTypes = {
  nodeData: PropTypes.shape({
    argument: PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.string.isRequired,
      })
    ).isRequired,
    options: PropTypes.shape({
      alt: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default Lightbox;
