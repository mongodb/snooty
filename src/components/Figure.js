import React, { Component } from 'react';
import { withPrefix } from 'gatsby';
import PropTypes from 'prop-types';
import CaptionLegend from './CaptionLegend';
import Lightbox from './Lightbox';
import { getNestedValue } from '../utils/get-nested-value';

export default class Figure extends Component {
  constructor(props) {
    super(props);
    this.imgRef = React.createRef();
    this.state = {
      isLightboxSize: false,
    };
  }

  componentDidMount() {
    const img = this.imgRef.current;
    if (img && img.complete) {
      this.handleImageLoaded();
    }
  }

  imgShouldHaveLightbox = () => {
    const img = this.imgRef.current;
    const naturalArea = img.naturalWidth * img.naturalHeight;
    const clientArea = img.clientWidth * img.clientHeight;
    return clientArea < naturalArea * 0.9;
  };

  handleImageLoaded = () => {
    this.setState({
      isLightboxSize: this.imgShouldHaveLightbox(),
    });
  };

  render() {
    const { nodeData, ...rest } = this.props;
    const { isLightboxSize } = this.state;
    const imgSrc = getNestedValue(['argument', 0, 'value'], nodeData);

    if (isLightboxSize || (nodeData.options && nodeData.options.lightbox)) {
      return <Lightbox nodeData={nodeData} />;
    }
    return (
      <div className="figure" style={{ width: getNestedValue(['options', 'figwidth'], nodeData) || 'auto' }}>
        <img
          src={withPrefix(imgSrc)}
          alt={getNestedValue(['options', 'alt'], nodeData) || imgSrc}
          width="50%"
          onLoad={this.handleImageLoaded}
          ref={this.imgRef}
        />
        <CaptionLegend {...rest} nodeData={nodeData} />
      </div>
    );
  }
}

Figure.propTypes = {
  nodeData: PropTypes.shape({
    argument: PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.string.isRequired,
      })
    ).isRequired,
    options: PropTypes.shape({
      alt: PropTypes.string,
      lightbox: PropTypes.bool,
    }).isRequired,
  }).isRequired,
};
