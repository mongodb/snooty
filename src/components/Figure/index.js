import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/react';
import Image from '../Image';
import { getNestedValue } from '../../utils/get-nested-value';
import { theme } from '../../theme/docsTheme';
import Lightbox from './Lightbox';
import CaptionLegend from './CaptionLegend';

const Figure = ({ ...props }) => {
  const figWidth = parseInt(getNestedValue(['nodeData', 'options', 'figwidth'], props), 10);
  const imgWidth = parseInt(getNestedValue(['nodeData', 'options', 'width'], props), 10);

  const [isLightboxSize] = useState(figWidth && figWidth / imgWidth < 0.9);

  const { nodeData, ...rest } = props;

  if (isLightboxSize || (nodeData.options && nodeData.options.lightbox)) {
    return <Lightbox nodeData={nodeData} />;
  }

  return (
    <div
      className="figure"
      css={css`
        max-width: 100%;
        margin-top: ${theme.size.medium};
        margin-bottom: ${theme.size.medium};
      `}
      style={{ width: getNestedValue(['options', 'figwidth'], nodeData) || 'auto' }}
    >
      <Image nodeData={nodeData} />
      <CaptionLegend {...rest} nodeData={nodeData} />
    </div>
  );
};

export default Figure;

Figure.propTypes = {
  nodeData: PropTypes.shape({
    argument: PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.string.isRequired,
      })
    ).isRequired,
    options: PropTypes.shape({
      alt: PropTypes.string,
      lightbox: PropTypes.bool,
      checksum: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
