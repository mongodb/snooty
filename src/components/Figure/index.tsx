import React, { useState } from 'react';
import { css, cx } from '@leafygreen-ui/emotion';
import Image from '../Image';
import { getNestedValue } from '../../utils/get-nested-value';
import { theme } from '../../theme/docsTheme';
import Lightbox from './Lightbox';
import CaptionLegend from './CaptionLegend';
import { FigureNode } from '../../types/ast';

export type FigureProps = {
  nodeData: FigureNode;
};

const Figure = ({ ...props }: FigureProps) => {
  const figWidth = parseInt(getNestedValue(['nodeData', 'options', 'figwidth'], props), 10);
  const imgWidth = parseInt(getNestedValue(['nodeData', 'options', 'width'], props), 10);

  const [isLightboxSize] = useState(figWidth && figWidth / imgWidth < 0.9);

  const { nodeData, ...rest } = props;

  if (isLightboxSize || (nodeData.options && nodeData.options.lightbox)) {
    return <Lightbox {...props} nodeData={nodeData} />;
  }

  return (
    <div
      className={cx(
        'figure',
        css`
          max-width: 100%;
          margin-top: ${theme.size.medium};
          margin-bottom: ${theme.size.medium};
        `
      )}
      style={{ width: getNestedValue(['options', 'figwidth'], nodeData) || 'auto' }}
    >
      <Image nodeData={nodeData} {...rest} />
      <CaptionLegend nodeData={nodeData} {...rest} />
    </div>
  );
};

export default Figure;
