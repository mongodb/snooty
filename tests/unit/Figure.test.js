import React from 'react';
import { render } from 'enzyme';
import Figure from '../../src/components/Figure';

// data for this component
import mockData from './data/Figure.test.json';
import borderData from './data/FigureBorder.test.json';
import lightboxData from './data/FigureLightbox.test.json';

it('renders correctly', () => {
  const tree = render(<Figure nodeData={mockData} />);
  expect(tree).toMatchSnapshot();
});

it('renders border correctly when specified as an option', () => {
  const tree = render(<Figure nodeData={borderData} />);
  expect(tree).toMatchSnapshot();
});

it('renders lightbox correctly when specified as an option', () => {
  const tree = render(<Figure nodeData={lightboxData} />);
  expect(tree).toMatchSnapshot();
});
