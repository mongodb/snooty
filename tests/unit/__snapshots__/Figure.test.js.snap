// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders border correctly when specified as an option 1`] = `
.emotion-1 {
  max-width: 100%;
}

.emotion-0 {
  border: 0.5px solid #B8C4C2;
  width: 100%;
  border-radius: 4px;
  max-width: 100%;
}

<div
  class="figure emotion-1"
  style="width:700px"
>
  <img
    alt="/images/firstcluster.png"
    class="   emotion-0"
    src="/images/firstcluster.png"
  />
</div>
`;

exports[`renders correctly 1`] = `
.emotion-1 {
  max-width: 100%;
}

.emotion-0 {
  max-width: 100%;
}

<div
  class="figure emotion-1"
  style="width:700px"
>
  <img
    alt="/images/firstcluster.png"
    class="   emotion-0"
    src="/images/firstcluster.png"
  />
</div>
`;

exports[`renders lightbox correctly when specified as an option 1`] = `
.emotion-0 {
  max-width: 100%;
}

<div
  class="figure lightbox"
  style="width:700px"
>
  <div
    class="lightbox__imageWrapper"
    role="button"
    tabindex="-1"
  >
    <img
      alt="/images/firstcluster.png"
      class="   emotion-0"
      src="/images/firstcluster.png"
    />
    <div
      class="lightbox__caption"
    >
      click to enlarge
    </div>
  </div>
</div>
`;
